commands/serve/web/components/property-panel/DataCube.jsx

```diff
@@ -27,25 +28,38 @@ const getValue = (data, reference, defaultValue) => {
 };
 
 export default function DataCube({ setProperties, target, properties }) {
-  const createHandler = target.propertyPath.match('/qHyperCube') ? hcHandler : loHandler;
+  // Store dimensions outside of handler to support filterpanes or other containers
+  // holding list objects, which do not store dimensions in an array (like hc.qDimensions).
+  const [dimensions, setDimensions] = useState([]);
+
+  const supportsMultipleDims = target.dimensions && target.dimensions.max && target.dimensions.max() > 1;
+  const listBoxHandler = supportsMultipleDims ? loContainerHandler : loHandler;
+
+  const createHandler = target.propertyPath.match('/qHyperCube') ? hcHandler : listBoxHandler;
+  const dc = getValue(properties, target.propertyPath);
   const handler = useMemo(
     () =>
       createHandler({
         def: target,
-        dc: getValue(properties, target.propertyPath),
+        dc,
         properties,
+        dimensions,
       }),
-    [properties]
+    [dc, target, properties, dimensions]
   );
 
   const onDimensionAdded = (a) => {
-    handler.addDimension(typeof a === 'object' ? { qLibraryId: a.qId } : a);
-    setProperties(properties);
+    const dim = typeof a === 'object' ? { qLibraryId: a.qId } : a;
+    const addedDim = handler.addDimension(dim);
+    setProperties(properties).then(() => {
+      setDimensions([...dimensions, addedDim]);
+    });
```

Comment:
Nor sure what this change does?

Reply:
It adds the dimension to the external dimensions array (`dimensions`) so that the Filter pane can handle multiple dimensions. This `dimensions` array is then sent in to the handler so it can return it whenever the property panel calls `getDImensions()`.

And it's chained so that it only does this if `setProperties` is successful.

Reply:
We may want to do the same (wait for success of `setProperties`) also when a dimension is removed, further down in this file.